hail/src/main/scala/is/hail/expr/ir/lowering/LoweringPipeline.scala

```diff
@@ -32,11 +27,83 @@ case class LoweringPipeline(lowerings: IndexedSeq[LoweringPass]) {
 }
 
 object LoweringPipeline {
-  val relationalLowerer: LoweringPipeline = LoweringPipeline(Array(LowerMatrixToTablePass, InterpretNonCompilablePass))
-  val legacyRelationalLowerer: LoweringPipeline = LoweringPipeline(Array(LowerMatrixToTablePass, LegacyInterpretNonCompilablePass))
-  val compileLowerer: LoweringPipeline = LoweringPipeline(Array(InlineApplyIR, LowerArrayAggsToRunAggsPass))
-  val darrayLowerer: Map[DArrayLowering.Type, LoweringPipeline] = Map(
-    DArrayLowering.All -> LoweringPipeline(Array(LowerMatrixToTablePass, LowerToDistributedArrayPass(DArrayLowering.All))),
-    DArrayLowering.TableOnly -> LoweringPipeline(Array(LowerMatrixToTablePass, LowerToDistributedArrayPass(DArrayLowering.TableOnly))),
-    DArrayLowering.BMOnly -> LoweringPipeline(Array(LowerToDistributedArrayPass(DArrayLowering.BMOnly))))
+  def relationalLowerer(optimize: Boolean): LoweringPipeline = if (optimize)
+    LoweringPipeline(
+      OptimizePass("relationalLowerer, initial IR"),
+      LowerMatrixToTablePass,
+      OptimizePass("relationalLowerer, after LowerMatrixToTable"),
+      InterpretNonCompilablePass,
+      OptimizePass("relationalLowerer, after InterpretNonCompilable")
+    )
+  else
+    LoweringPipeline(
+      LowerMatrixToTablePass,
+      InterpretNonCompilablePass
+    )
```

Comment:
An idea to simplify the pipeline definitions: you could add a method on `LoweringPipeline` to drop all optimization passes. Then you could just define the pass once, and return `if (optimize) pass else pass.noOptimization`. For a cross cutting concern like whether to optimize between every other pass, I like that better. But if you prefer this, I'm fine with it.

Reply:
oh, definitely better, thanks for the suggestion. That's why I non-randomly assigned you 😉